sdk/cosmos/azure-cosmos/azure/cosmos/cosmos_diagnostics.py

```diff
@@ -0,0 +1,265 @@
+
+    @status_code.setter
+    def status_code(self, value: int):
+        self._status_code = value
+
+    @property
+    def substatus_code(self):
+        return self._substatus_code
+
+    @substatus_code.setter
+    def substatus_code(self, value: int):
+        self._substatus_code = value
+
+    @property
+    def status_reason(self):
+        return self._status_reason
+
+    @status_reason.setter
+    def status_reason(self, value: str):
+        self._status_reason = value
+
+    @property
+    def elapsed_time(self):
+        return self._elapsed_time
+
+    @elapsed_time.setter
+    def elapsed_time(self, value):
+        self._elapsed_time = value
+
+    @property
+    def status_message(self):
+        return self._status_message
+
+    @status_message.setter
+    def status_message(self, value: str):
+        self._status_message = value
+
+    @property
+    def user_agent(self):
+        return self._user_agent
+
+    @user_agent.setter
+    def user_agent(self, value: str):
+        self._user_agent = value
+
+    @property
+    def consistency_level(self):
+        return self._consistency_level
+
+    @consistency_level.setter
+    def consistency_level(self, value: str):
+        self._consistency_level = value
+
+    @property
+    def operation_type(self):
+        return self._operation_type
+
+    @operation_type.setter
+    def operation_type(self, value: str):
+        self._operation_type = value
+
+    @property
+    def resource_type(self):
+        return self._resource_type
+
+    @resource_type.setter
+    def resource_type(self, value: str):
+        self._resource_type = value
+
+    def update_diagnostics(self, header: dict, body: dict, **kwargs):
+        self.clear()
+        self.response_headers = header
+        try:
+            self.body = dict(body)
+        except ValueError as ve:
+            self.body = body
+        eTime = kwargs.get("elapsed_time")
+        if eTime:
+            self.elapsed_time = eTime
+        e = kwargs.get("exception")
+        sc = kwargs.get("status_code")
+        if sc:
+            self.status_code = sc
+        sr = kwargs.get("status_reason")
+        if sr:
+            self.status_reason = sr
+        sm = kwargs.get("response_text")
+        if sm:
+            self.status_message = sm
+        ssc = kwargs.get("substatus_code")
+        if ssc:
+            self.substatus_code = ssc
+        else:
+            self.substatus_code = 0
+        ua = kwargs.get("user_agent")
+        if ua:
+            self.user_agent = ua
+        rh = kwargs.get("request_headers")
+        if rh:
+            self.request_headers = dict(rh)
+        ot = kwargs.get("operation_type")
+        if ot:
+            self.operation_type = ot
+        rt = kwargs.get("resource_type")
+        if rt:
+            self.resource_type = rt
+        if e:
+            self.status_code = e.status_code
+            self.status_message = e.message
+
+    def __call__(self, p=False):
```

Comment:
I know you were probably attempting to be less verbose but let's please make the 'p' variable something understandable like 'print' or 'print_out'

Reply:
I agree, I will make that change. I believe print_out should work better.